toolchain/sem_ir/entity_name.h

```diff
@@ -53,9 +53,11 @@ struct EntityName : public Printable<EntityName> {
   // them for other kinds of `EntityName`.
 
   // The bind_index() value, unwrapped so it can be stored in a bit-field.
-  int32_t bind_index_value : 31 = CompileTimeBindIndex::None.index;
+  int32_t bind_index_value : 30 = CompileTimeBindIndex::None.index;
```

Comment:
I'm hesitant about further splitting of this index. It may be fine, but had you considered adding a `CARBON_CHECK` that the input value fits? That way, a too-large value would be a clearer error.

Reply:
Done.